context/app/static/js/components/files/FileBrowser/FileBrowser.spec.js

```diff
@@ -59,13 +62,75 @@ test('displays files and directories', () => {
   );
 
   const textInDocumentBeforeOpenDirectory = ['path1', 'path3', 'fake5.txt'];
-  textInDocumentBeforeOpenDirectory.forEach((text) => expect(screen.getByText(text)).toBeInTheDocument());
+  expectArrayOfStringsToExist(textInDocumentBeforeOpenDirectory);
 
   const textNotInDocumentBeforeOpenDirectory = ['path2', 'fake1.txt', 'fake2.txt', 'fake3.txt', 'fake4.txt'];
-  textNotInDocumentBeforeOpenDirectory.forEach((text) => expect(screen.queryByText(text)).toBeNull());
+  expectArrayOfStringsToNotExist(textNotInDocumentBeforeOpenDirectory);
 
   userEvent.click(screen.getByRole('button', { name: 'path1' }));
 
   const textInDocumentAfterOpenDirectory = [...textInDocumentBeforeOpenDirectory, 'fake3.txt', 'path2'];
-  textInDocumentAfterOpenDirectory.forEach((text) => expect(screen.getByText(text)).toBeInTheDocument());
+  expectArrayOfStringsToExist(textInDocumentAfterOpenDirectory);
+});
+
+test('Displays correct files before and after display only QA files chip is clicked', () => {
+  const sharedEntries = {
+    edam_term: 'faketerm',
+    description: 'fakedescription',
+    size: 1000,
+    type: 'faketype',
+  };
+
+  const testFiles = [
+    {
+      rel_path: 'path1/path2/fake1.txt',
+      ...sharedEntries,
+      is_qa_qc: true,
+    },
+    {
+      rel_path: 'path1/path2/fake2.txt',
+      ...sharedEntries,
+    },
+    {
+      rel_path: 'path1/fake3.txt',
+      ...sharedEntries,
+      is_qa_qc: true,
+    },
+    {
+      rel_path: 'path3/fake4.txt',
+      ...sharedEntries,
+    },
+    {
+      rel_path: 'fake5.txt',
+      ...sharedEntries,
+    },
+  ];
+
+  render(
+    <FilesProviders>
+      <FileBrowser files={testFiles} />
+    </FilesProviders>,
+  );
+
+  // intial
+  const textInDocumentBeforeQAFilesOnly = ['path1', 'path3', 'fake5.txt'];
+  expectArrayOfStringsToExist(textInDocumentBeforeQAFilesOnly);
+
+  const textNotInDocumentBeforeQAFilesOnly = ['path2', 'fake1.txt', 'fake2.txt', 'fake3.txt', 'fake4.txt'];
+  expectArrayOfStringsToNotExist(textNotInDocumentBeforeQAFilesOnly);
+
+  userEvent.click(screen.getByRole('button', { name: 'Show QA Files Only' }));
+
+  // after QA files chip clicked
+  const textInDocumentAfterQAFilesOnly = ['path1', 'path2', 'fake1.txt', 'fake3.txt'];
+  expectArrayOfStringsToExist(textInDocumentAfterQAFilesOnly);
+
+  const textNotInDocumentAfterQAFilesOnly = ['path3', 'fake4.txt', 'fake2.txt', 'fake5.txt'];
+  expectArrayOfStringsToNotExist(textNotInDocumentAfterQAFilesOnly);
+
+  userEvent.click(screen.getByRole('button', { name: 'Show QA Files Only' }));
+
+  // returns to all dirs closes
+  expectArrayOfStringsToExist(textInDocumentBeforeQAFilesOnly);
+  expectArrayOfStringsToNotExist(textNotInDocumentBeforeQAFilesOnly);
```

Comment:
Cool. I've sometimes put more semantics into the mocks... ie, `file-to-hide-unless-button-pressed.txt` or something like that. If assertions fail in the future, it can tell immediately what went wrong... but there's the risk of paying too much attention to the names, and not the criteria that actually matter.

Just a thought. Your call.

Reply:
I'll try to do so in the future, but I'll leave it as is here unless you think it makes sense to incorporate now.